src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -629,6 +657,41 @@ uct_dc_mlx5_ep_flush_cancel(uct_dc_mlx5_ep_t *ep)
     return UCS_INPROGRESS;
 }
 
+static uint32_t uct_dc_mlx5_flush_remote_rkey(uct_dc_mlx5_ep_t *ep)
+{
+    ucs_assert((ep->atomic_mr_offset & 0xff) == 0);
+    return ((uint32_t)ep->flush_remote_rkey << 16) + ep->atomic_mr_offset;
+}
+
+static ucs_status_t
+uct_dc_mlx5_ep_flush_remote(uct_dc_mlx5_ep_t *ep, uct_completion_t *comp)
+{
+    uct_dc_mlx5_iface_t *iface = ucs_derived_of(ep->super.super.iface,
+                                                uct_dc_mlx5_iface_t);
+    uint32_t flush_remote_rkey = uct_dc_mlx5_flush_remote_rkey(ep);
+    uct_rc_iface_send_desc_t *desc;
+    UCT_DC_MLX5_TXQP_DECL(txqp, txwq);
+
+    UCT_DC_MLX5_CHECK_RES(iface, ep);
+
+    UCT_RC_IFACE_GET_TX_DESC(iface, &iface->super.super.tx.mp, desc);
+    desc->super.handler   = uct_rc_ep_flush_remote_handler;
+    desc->super.user_comp = comp;
+
+    UCT_DC_MLX5_IFACE_TXQP_GET(iface, ep, txqp, txwq);
+
+    uct_dc_mlx5_iface_bcopy_post(iface, ep, MLX5_OPCODE_RDMA_READ,
+                                 UCT_RC_IFACE_FLUSH_REMOTE_LENGTH, 0,
+                                 flush_remote_rkey, desc, 0, 0, desc + 1, NULL);
+
+    UCT_RC_RDMA_READ_POSTED(&iface->super.super,
```

Comment:
should we really count flow control resources for flush-remote operations?
I think not